linode/stackscript/framework_models.go

```diff
@@ -32,17 +32,46 @@ type StackScriptModel struct {
 	UserDefinedFields basetypes.ListValue `tfsdk:"user_defined_fields"`
 }
 
-func (data *StackScriptModel) parseStackScript(
+func (data *StackScriptModel) parseNonComputedAttributes(
 	ctx context.Context,
 	stackscript *linodego.Stackscript,
 ) diag.Diagnostics {
-	data.ID = types.StringValue(strconv.Itoa(stackscript.ID))
+	var diagnostics diag.Diagnostics
+
 	data.Label = types.StringValue(stackscript.Label)
 	data.Script = types.StringValue(stackscript.Script)
 	data.Description = types.StringValue(stackscript.Description)
-	data.RevNote = helper.GetValueIfNotNull(stackscript.RevNote)
+
+	// These fields have schema-defined defaults,
+	// so this should not cause comparison issues
+	data.RevNote = types.StringValue(stackscript.RevNote)
 	data.IsPublic = types.BoolValue(stackscript.IsPublic)
 
+	// Only update the images return if there is a change
+	var plannedImages []string
+	diagnostics.Append(data.Images.ElementsAs(ctx, &plannedImages, false)...)
+	if diagnostics.HasError() {
+		return diagnostics
+	}
+
+	if !helper.StringListElementsEqual(plannedImages, stackscript.Images) {
+		remoteImages, err := types.ListValueFrom(ctx, types.StringType, stackscript.Images)
+		if err != nil {
+			return err
```

Comment:
```suggestion
		remoteImages, diags := types.ListValueFrom(ctx, types.StringType, stackscript.Images)
		if diags.HasError() {
```

Reply:
There may be this type of thing elsewhere. Can you screen all `err != nil` to figure out if there should be `diags` instead?